lib/ii.c

```diff
@@ -11758,8 +11763,10 @@ grn_ii_select_data_check_near_element_intervals(grn_ctx *ctx,
       int32_t pos = data->check_element_intervals_btree->min->pos;
       int32_t max_element_interval =
         GRN_INT32_VALUE_AT(data->max_element_intervals, i - 1);
-      if (max_element_interval >= 0 &&
-          (pos - previous_pos) > max_element_interval) {
+      int32_t interval = pos - previous_pos;
+      if ((max_element_interval >= 0 &&
+           interval > max_element_interval) ||
+          interval < data->min_interval) {
         return false;
       }
```

Comment:
Could you use 2 `if {return false;}` for readability?

```c
if (max_element_interval >= 0 && interval > max_element_interval) {
  return false;
}
if (interval < data->min_interval) {
  return false;
}
```

Reply:
Fixed